decorator.rb

```diff
@@ -0,0 +1,28 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(namable)
+    super()
+    @namable = namable
+  end
+
+  def correct_name
+    @namable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if super.length > 10
+      super[0..9]
+    else
+      super
+    end
+  end
```

Comment:
- [ ] **[OPTIONAL]** Based on Ruby conventions classes should be in their own files. I highly recommend you move this class to its own file.